src/main/scala/com/github/unchama/seichiassist/subsystems/seichilevelupgift/usecases/GrantGiftOnSeichiLevelDiff.scala

```diff
@@ -0,0 +1,52 @@
+package com.github.unchama.seichiassist.subsystems.seichilevelupgift.usecases
+
+import cats.Applicative
+import cats.effect.Bracket.catsKleisliBracket
+import cats.effect.Sync
+import com.github.unchama.generic.Diff
+import com.github.unchama.generic.algebra.typeclasses.HasSuccessor
+import com.github.unchama.minecraft.actions.SendMinecraftMessage
+import com.github.unchama.seichiassist.subsystems.breakcount.domain.level.SeichiLevel
+import com.github.unchama.seichiassist.subsystems.seichilevelupgift.domain.{
+  Gift,
+  GiftBundleTable,
+  GrantLevelUpGift
+}
+
+class GrantGiftOnSeichiLevelDiff[F[_], Player] {
+
+  import cats.implicits._
+
+  final def grantGiftTo(levelDiff: Diff[SeichiLevel], player: Player)(
```

Comment:
```suggestion
object GrantGiftOnSeichiLevelDiff {

  import cats.implicits._

  def grantGiftTo[F[_], Player](levelDiff: Diff[SeichiLevel], player: Player)(
```

としてよさそう、わざわざ `new` する意味があんまり無さそう